src/xls.rs

```diff
@@ -464,57 +464,53 @@ impl<RS: Read + Seek> Xls<RS> {
 /// https://learn.microsoft.com/en-us/openspecs/office_file_formats/ms-xls/4d6a3d1e-d7c5-405f-bbae-d01e9cb79366
 struct Bof {
     /// Binary Interchange File Format
-    biff: u8,
+    biff: Biff,
+}
+
+/// https://www.loc.gov/preservation/digital/formats/fdd/fdd000510.shtml#notes
+enum Biff {
+    Biff2,
+    Biff3,
+    Biff4,
+    Biff5,
+    Biff8,
+    // Used by MS-XLSB Workbook(2.1.7.61) or Worksheet(2.1.7.61) which are not supported yet.
+    // Biff12,
 }
 
 /// BOF [MS-XLS] 2.4.21
 fn parse_bof(r: &mut Record<'_>) -> Result<Bof, XlsError> {
     let mut dt = 0;
     let biff_version = read_u16(&r.data[..2]);
 
-    if r.data.len() >= 2 {
+    if r.data.len() >= 4 {
         dt = read_u16(&r.data[2..]);
     };
 
-    let mut biff = match biff_version {
-        // BIFF8
-        0x0600 => 8,
-        // BIFF5
-        0x0500 => 5,
-        // BIFF4
-        0x0400 => 4,
-        // BIFF3
-        0x0300 => 3,
-        // BIFF2
-        0x0200 | 0x0002 | 0x0007 => 2,
-        _ => {
-            if r.data[4..].len() > 6 {
-                return Err(XlsError::Len {
-                    typ: "BOF data length",
-                    expected: 6,
-                    found: r.data.len(),
-                });
+    let biff = match biff_version {
+        0x0200 | 0x0002 | 0x0007 => Biff::Biff2,
+        0x0300 => Biff::Biff3,
+        0x0400 => Biff::Biff4,
+        0x0500 => Biff::Biff5,
+        0x0600 => Biff::Biff8,
+        0 => {
+            if dt == 0x1000 {
+                Biff::Biff5
             } else {
-                8
+                Biff::Biff8
             }
         }
+        _ => Biff::Biff8,
     };
 
-    if biff_version == 0 && dt == 0x1000 {
-        biff = 5;
-    }
-    if biff == 8 && biff_version == 0 && dt == 16 {
```

Comment:
It looks like this logic was used for a weird file (https://github.com/SheetJS/sheetjs/issues/1220) and I'm not sure we need/want to support this since I cannot really find a test file for it. I'm still trying to understand the reason for `dt == 0x1000`.